### PR TITLE
Capture headers discussion from 2020-03-26 SDK meeting

### PR DESCRIPTION
**Pull request by BillyONeal**

### CODE REVIEW

File: docs/cpp/implementation.md [hunk 1]
```diff
@@ -319,6 +319,14 @@ Filenames should be concise, but convey what role the file plays within the libr
 // Contents of a given header
 {% endhighlight %}
 
+{% include requirement/MUST id="cpp-style-whole-sdk-header" %} have a header file that includes an entire SDK. For example, `azure_storage.h` or `<azure/storage.h>`.
+
+> TODO: should directory structure mimic the namespace structure with directories or all in one directory with underscores?
```
Comment (gilbertw) [inline docs/cpp/implementation.md]: Directory structure matching the namespace gets my vote.

**Comment by BillyONeal** [comment]
The recent update also captures the .hpp distinction.

File: docs/cpp/implementation.md [hunk 2]
```diff
@@ -301,11 +301,9 @@ TEST_FUNCTION(foo_tcp_manager_create_createAndReturnInstanceSucceed)
...
+{% include requirement/MUST id="cpp-style-filenaming" %} name all files as lowercase, in a directory of the service short name. Separate words with underscores, and end with the appropriate extension (`.cpp` or `.hpp`). For example, `iot_credential.cpp` is valid, while `IoTCredential.cl` is not.
 
-{% include requirement/MUST id="cpp-style-publicapi-hdr" %} identify the file containing the public API with `<svcname>_<objname>_api.h`.  For example, `iot_credential_api.h`.
-
-{% include requirement/MUST id="cpp-style-privateapi-hdr" %} place an include file that is not part of the public API in an `internal` directory.  Do not include the service short name.  For example, `internal/credential.h`.
+{% include requirement/MUST id="cpp-style-privateapi-hdr" %} place an include file that is not part of the public API in an `internal` directory.  Do not include the service short name.  For example, `<azure/internal/credential.hpp>`.
```
Comment (gilbertw) [inline docs/cpp/implementation.md:307]: I would expect internal headers for a service to be in service directory.  That's not clear from the above.

Additionally should we mandate the includes are under the inc folder?

**Comment by BillyONeal** [comment]
@gilbertw I'm not sure, can you put it on the agenda for next week?

**Pull request merged**